packages/ra-core/src/controller/list/useList.storeKey.stories.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import { Route } from 'react-router';
+import { Link } from 'react-router-dom';
+import fakeDataProvider from 'ra-data-fakerest';
+
+import {
+    CoreAdminContext,
+    CoreAdminUI,
+    CustomRoutes,
+    Resource,
+} from '../../core';
+import { localStorageStore } from '../../store';
+import { FakeBrowserDecorator } from '../../storybook/FakeBrowser';
+import { CoreLayoutProps, SortPayload } from '../../types';
+import { useList } from './useList';
+
+export default {
+    title: 'ra-core/controller/list/useList',
+    decorators: [FakeBrowserDecorator],
+    parameters: {
+        initialEntries: ['/top'],
+    },
+};
+
+const styles = {
+    mainContainer: {
+        margin: '20px 10px',
+    },
+    ul: {
+        marginTop: '20px',
+        padding: '10px',
+    },
+};
+
+const dataProvider = fakeDataProvider({
+    posts: [
+        { id: 1, title: 'Post #1', votes: 90 },
+        { id: 2, title: 'Post #2', votes: 20 },
+        { id: 3, title: 'Post #3', votes: 30 },
+        { id: 4, title: 'Post #4', votes: 40 },
+        { id: 5, title: 'Post #5', votes: 50 },
+        { id: 6, title: 'Post #6', votes: 60 },
+        { id: 7, title: 'Post #7', votes: 70 },
+    ],
+});
+
+const OrderedPostList = ({
+    storeKey,
+    sort,
+}: {
+    storeKey: string | false;
+    sort?: SortPayload;
+}) => {
+    const params = useList({
+        resource: 'posts',
+        perPage: 3,
+        sort,
+        storeKey,
+    });
+
+    return (
+        <div>
+            <span aria-label="storeKey" data-value={storeKey}>
+                storeKey: {storeKey}
+            </span>
+            <br />
+            <span aria-label="perPage" data-value={params.perPage}>
+                perPage: {params.perPage}
+            </span>
+            <br />
+            <button
+                aria-label="incrementPerPage"
+                onClick={() => {
+                    return params.setPerPage(++params.perPage);
+                }}
+            >
+                Increment perPage
+            </button>
+            <ul style={styles.ul}>
+                {params.data?.map(post => (
+                    <li key={post.id}>
+                        {post.title} - {post.votes} votes
+                    </li>
+                ))}
+            </ul>
+        </div>
+    );
+};
+
+const Layout = (props: CoreLayoutProps) => (
+    <div style={styles.mainContainer}>
+        <Link aria-label="top" to={`/top`}>
+            Go to Top Posts
+        </Link>
+        <Link aria-label="flop" to={`/flop`}>
+            Go to Flop Posts
+        </Link>
+        <Link aria-label="store" to={`/store`}>
+            Go to Store List
+        </Link>
+        <Link aria-label="nostore" to={`/nostore`}>
+            Go to No-Store List
+        </Link>
+
+        <br />
+        {props.children}
+    </div>
+);
```

Comment:
`Layout` should not be shared across stories since both stories do not implement the same routes